src/image_editing.rs

```diff
@@ -921,215 +1116,293 @@ impl ImageOperation {
     }
 
     /// Process all image operators (All things that modify the image and are not "per pixel")
-    pub fn process_image(&self, img: &mut RgbaImage) -> Result<()> {
-        match self {
-            Self::Blur(amt) => {
-                if *amt != 0 {
-                    let i = img.clone();
-                    let mut data = i.into_raw();
-                    libblur::stack_blur(
-                        data.as_mut_slice(),
-                        img.width() * 4,
-                        img.width(),
-                        img.height(),
-                        (*amt as u32).clamp(2, 254),
-                        libblur::FastBlurChannels::Channels4,
-                        libblur::ThreadingPolicy::Adaptive,
-                    );
-                    use anyhow::Context;
-                    *img = RgbaImage::from_raw(img.width(), img.height(), data)
-                        .context("Can't construct image from blur result")?;
-                }
-            }
-            Self::Filter3x3(amt) => {
-                let kernel = amt
-                    .into_iter()
-                    .map(|a| *a as f32 / 100.)
-                    .collect::<Vec<_>>();
-                *img = imageops::filter3x3(img, &kernel);
-            }
-            Self::LUT(lut_name) => {
-                use lutgen::identity::correct_image;
-                let mut external_image = DynamicImage::ImageRgba8(img.clone()).to_rgb8();
-                if let Some(lut_data) = builtin_luts().get(lut_name) {
-                    let lut_img = image::load_from_memory(&lut_data).unwrap().to_rgb8();
-                    correct_image(&mut external_image, &lut_img);
-                } else {
-                    if let Ok(lut_img) = image::open(&lut_name) {
-                        correct_image(&mut external_image, &lut_img.to_rgb8());
+    pub fn process_image(&self, dyn_img: &mut DynamicImage) -> Result<()> {
+        match dyn_img {
+            DynamicImage::ImageRgba8(img) => {
+                match self {
+                    Self::Blur(amt) => {
+                        if *amt != 0 {
+                            let i = img.clone();
+                            let mut data = i.into_raw();
+                            libblur::stack_blur(
+                                data.as_mut_slice(),
+                                img.width() * 4,
+                                img.width(),
+                                img.height(),
+                                (*amt as u32).clamp(2, 254),
+                                libblur::FastBlurChannels::Channels4,
+                                libblur::ThreadingPolicy::Adaptive,
```

Comment:
libblur's have gated feature _image_ with prebuilt handling `DynamicImage` if this makes sense in latest versions
https://github.com/awxkee/libblur/blob/03341a4e4df05d88bf61f69187cbe2c16cee7770/src/lib/stack_blur_image.rs#L46.

Just to confirm because it is not clear from context: it is a mistake that immediately leads to artifacts to perform blur/scaling/any blending operation on the image with unpremultiplied alpha. The `DynamicImage` type from `image` crate always considers an image as unpremultiplied, therefore to perform any blending operation you have to premultiply alpha first. Thus this leads to question, does here `DynamicImage` holds an image with already premultiplied alpha?